lib/web3-contracts.js

```diff
@@ -61,6 +42,36 @@ export function useTokenDecimals(symbol) {
   return tokenDecimals.get(symbol)
 }
 
+export function useConnectorWeight() {
+  const marketMakerContract = useKnownContract('MARKET_MAKER')
+  const [address] = getKnownContract('COLLATERAL_TOKEN')
+  
+  const [reserveRatio, setReserveRatio] = useState()
+  const [collateralToken, setCollateralToken] = useState()
+
+  const fetchCollateral = useCallback(async () => {
+    try {
+      if(!marketMakerContract) {
+        return null
+      }
+      const collateralToken = await marketMakerContract.getCollateralToken(address)
+  
+      setCollateralToken(collateralToken)
+  
+      return setReserveRatio(collateralToken[collateralToken.length - 1])
```

Comment:
On the top of these lines we could add what this function returns, something like:
```
// getCollateralToken(collateral: address): [whitelisted: bool, virtualSupply: uint256, virtualBalance: uint256, reserveRatio: uint32]
```

I think it's important to replace `collateralToken.length - 1` by just `3` to indicate that it's the argument.

So I would replace these three lines with:
```
const RESERVE_RATIO_ARG = 3
return setReserveRatio((await marketMakerContract.getCollateralToken(address))[RESERVE_RATIO_ARG])
```
That would be preferable but I'm not sure if it's really possible with what the function returns:
```
const [,,, reserveRatio] = await marketMakerContract.getCollateralToken(address)
return setReserveRatio(reserveRatio)
```